overlord/configstate/configcore/services.go

```diff
@@ -90,6 +91,48 @@ func switchDisableSSHService(sysd systemd.Systemd, serviceName, value string, op
 	}
 }
 
+// switchDisableSSHService handles the special case of disabling/enabling
+// console-conf on core devices.
+//
+// The only command that works to start/stop console-conf after setting
+// the marker file in /var/lib/console-conf/complete is:
+//
+//     systemctl restart 'getty@*' --all
+//
+func switchDisableConsoleConfService(sysd systemd.Systemd, serviceName, value string, opts *fsOnlyContext) error {
+	rootDir := dirs.GlobalRootDir
+	if opts != nil {
+		rootDir = opts.RootDir
+	}
+	if err := os.MkdirAll(filepath.Join(rootDir, "/var/lib/console-conf"), 0755); err != nil {
+		return err
+	}
+
+	consoleConfCanary := filepath.Join(rootDir, "/var/lib/console-conf/complete")
```

Comment:
console-conf service has:
```
[Unit]
ConditionPathExists=/dev/tty0
ConditionPathExists=!/var/lib/console-conf/complete

[Service]
Environment=PYTHONPATH=/usr/share/subiquity
ExecStartPre=/bin/systemctl stop getty@%I
ExecStart=/sbin/agetty -i -n --noclear -l /usr/share/subiquity/console-conf-wrapper %I $TERM
ExecStopPost=/bin/systemctl start getty@%I
...
```

The chooser is invoked from console-conf-wrapper. With this we get the ability to have the chooser appear on every terminal or serial console that console-conf has access to. If the service does not execute then we'll not start the chooser even if triggered. Perhaps we should land a fix in subiquity first, try to run console-conf at least once, even if /var/lib/console-conf/complete exists, unless tehre is a way to express that via systemd unit conditions. Perhaps `ConditionPathExists=|/run/snapd-recovery-chooser-triggered` will do the trick?

Reply:
>  Perhaps we should land a fix in subiquity first, try to run console-conf at least once, even if /var/lib/console-conf/complete exists, unless tehre is a way to express that via systemd unit conditions. Perhaps ConditionPathExists=|/run/snapd-recovery-chooser-triggered will do the trick?

I'm confused, does it mean that once you have run console-conf you cannot run the chooser anymore? that sounds wrong

Reply:
Aha, excellent point, I forgot we invoke inside console-conf-wrapper, so yes, this is blocked. It sounds like we always want to run the console-conf unit and instead change the console-conf-wrapper to check for the "complete" stamp there?

Reply:
Yes, I'll open a PR to subiquity with the fix.

Reply:
I've opened https://github.com/CanonicalLtd/subiquity/pull/770 with the fix. Now, it's a matter if getting it into core20.

Reply:
This got merged now in subiquity.